src/osltoy/codeeditor.cpp

```diff
@@ -68,6 +68,25 @@ CodeEditor::CodeEditor(QWidget* parent, const std::string& filename)
 
 
 
+void
+CodeEditor::keyPressEvent(QKeyEvent* event)
+{
+    if (event->key() == Qt::Key_Tab) {
+        // replace the tab key with 4 spaces
+        // TODO: make this a configurable preference?
+        auto replacement
+            = QKeyEvent(QEvent::KeyPress, Qt::Key_Space,
+                        Qt::KeyboardModifiers(event->nativeModifiers()),
+                        "    ");
```

Comment:
Hmmm, in most editors, does TAB do the equivalent of hitting SPACE 4 times? Or does it do the number of spaces to take you to the next 4-space boundary?

Reply:
Good point, I'm not sure why I didn't think of that. I'll take another look when I have a moment

Reply:
Ping just in case you're interested in fixing this up.